storage/src/mmr/hasher.rs

```diff
@@ -1,40 +1,76 @@
 //! Decorator for a cryptographic hasher that implements the MMR-specific hashing logic.
 
+use std::future::Future;
+
+use crate::mmr::{Error, Storage};
 use commonware_cryptography::Hasher as CHasher;
 
+/// A trait required by various MMR methods for computing hash digests.
+pub trait Hasher<H: CHasher>: Send + Sync {
+    /// Computes the hash for a leaf given its position and the element it represents.
```

Comment:
consistency: we refer to the output being a `digest` rather than a `hash` (but call the action of producing a `digest` `hashing`).

Reply:
changed